java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspTemplateUI.java

```diff
@@ -229,9 +254,18 @@ private static CompletionStage<FileBuilder> configure(FileBuilder builder, NbCod
         FileObject template = desc.getTemplate();
         Object handler = template.getAttribute(FileBuilder.ATTR_TEMPLATE_HANDLER);
         if (handler == null) {
-            return client.showInputBox(new ShowInputBoxParams(Bundle.CTL_TemplateUI_SelectName(), desc.getProposedName())).thenApply(name -> {
-                return name != null ? builder.name(name) : null;
-            });
+            class ValidateObjectName implements Function<String, CompletionStage<String>> {
+
+                @Override
+                public CompletionStage<String> apply(String name) {
+                    if (!SourceVersion.isName(name)) {
```

Comment:
This may be OK for Java templates, but probably not for others (like plain files, XML/HTML files, etc.) And this will be used for all file types, I think. I don't know offhand how to do validation only for Java, it might need another attribute on the template(?). @sdedic? Do I miss something?

Reply:
> I don't know offhand how to do validation only for Java, it might need another attribute on the template(?). @sdedic? Do I miss something?

I think a validator could specified by a template's file attribute, or by template's MIME type (for singe-file templates).

Reply:
@lahodaj  Thanks for pointing that out ! I looked up a bit more on this , it seems that all java templates have names ending with ".java" suffix  , Can you confirm that ? If thats the case then I can condition on the template name whether to apply the validation check or not .

Reply:
Updated the PR with conditional validation using MIME type .